### PR TITLE
Move all Helm charts off of stable, migrate dcgm-exporter to nvcr.io, bump versions of metallb/ingress/dcgm-exporter/prometheus

### PR DESCRIPTION
**Pull request by supertetelman**
* Use NGC instead of DockerHub for DCGM exporter container
* Migrate ingress-nginx, prometheus-operator, metallb, off of deprecated stable helm repo and use new community supported locations
* Fix issue in monitoring test where it passed on failing scenarios (404s and timeouts)
* Change naming of installs appropriately
* Bump to **dcgm-exporter** version to **2.0 RC**
* Bump prometheus-operator (**kube-prometheus-stack**) to **9.4.10**
* Bump/peg **metallb** to **0.1.24**
* Bump **ingress-nginx** (previuosly nginx-ingress) to **3.5.1**
* Automatically reconfigure ingress to use NodePort/LoadBalancer as available

**Manual Upgrade Steps**
Depending on the previous installs, you may need to manually delete some left-behind objects by running:
```sh
kubectl delete psp -l release=kube-prometheus-stack -A
kubectl delete psp -l release=prometheus-operator -A
kubectl delete psp -l app.kubernetes.io/instance=kube-prometheus-stack -A
kubectl delete psp -l app.kubernetes.io/instance=prometheus-operator -A
kubectl delete ClusterRole -l app=prometheus-operator-prometheus
kubectl delete ClusterRole -l app.kubernetes.io/instance=kube-prometheus-stack
kubectl delete ClusterRole -l app=kube-prometheus-stack-operator
kubectl delete ClusterRole -l  app=kube-prometheus-stack-prometheus
kubectl delete ClusterRole -l  app=prometheus-operator-operator
kubectl delete ClusterRole -l  app.kubernetes.io/instance=prometheus-operator
kubectl delete ClusterRole -l  app=prometheus-node-exporter
kubectl delete ClusterRoleBinding -l app=prometheus-operator-prometheus
kubectl delete ClusterRoleBinding -l app.kubernetes.io/instance=kube-prometheus-stack
kubectl delete ClusterRoleBinding -l app=kube-prometheus-stack-operator
kubectl delete ClusterRoleBinding -l  app=kube-prometheus-stack-prometheus
kubectl delete ClusterRoleBinding -l  app=prometheus-operator-operator
kubectl delete ClusterRoleBinding -l  app.kubernetes.io/instance=prometheus-operator
kubectl delete ClusterRoleBinding -l  app=prometheus-node-exporter
kubectl delete MutatingWebhookConfiguration -l release=kube-prometheus-stack
kubectl delete MutatingWebhookConfiguration -l release=prometheus-operator
kubectl delete MutatingWebhookConfiguration -l app=kube-prometheus-stack-admission
kubectl delete ValidatingWebhookConfiguration -l app=kube-prometheus-stack-admission
kubectl delete service -n kube-system $(kubectl get service -A -n kube-system | grep -e kube-prometheus-stack -e prometheus-operator | awk '{print $2}')
```

**Note: I didn't bother updating the airgap build offline script, as we dropped support and this needs a rehaul anyways.**

### CODE REVIEW

**Comment by supertetelman** [comment]
Before merging this I notcied that both `kubectl wait` and `helm install --wait` seem to hang and eventually timeout for no clear reason. In this scripts it will run until it hits the timeout, but if you open a new terminal and manually run the same command, the `Ready` condition is met and the wait immediately returns. Not sure if this is a Helm issue or an implementation issue.

**Comment by supertetelman** [comment]
I've got this running now but there are three things that need to be addressed.

1. The upgrade from prometheus-operator to the new kube-prometheus-stack requires removing the old operator and ingress. When I do this I am running into the below error during upgrade/install,  it seems like something isn't getting cleaned up and it is still referencing the old operator.
```
I might need your help looking into this as I also was seeing a potentially related (or a missed file) error Error: Internal error occurred: failed calling webhook "prometheusrulemutate.monitoring.coreos.com": Post https://prometheus-operator-operator.monitoring.svc:443/admission-prometheusrules/validate?timeout=30s: service "prometheus-operator-operator" not found
```

2. The docs still need to be updated.

3. The default configuration for ingress is `LoadBalancer`. This was not previously a problem even on systems with no LoadBalancer, the helm install was still running fine. However, now on my local systems that do not have a load balancer, the helm install changes and I had to manually change this. I'm not sure if we want to leave things as is, change our default configuration, or add some logic to our `deploy_ingress.sh` to detect metallb and reconfigure ingress appropriately.

**Review by ajdecon — CHANGES_REQUESTED** [review]
Overall this looks good, only making one request for a change so that we aren't editing the user's configuration. See inline comment.

**Review by michael-balint — COMMENTED** [review]
Tested - please address @ajdecon's suggestions so we can approve + merge.

File: scripts/k8s/deploy_ingress.sh [hunk 1]
```diff
@@ -38,9 +40,20 @@ if [ "${NGINX_INGRESS_BACKEND_REPO}" ]; then
 	helm_arguments+=("--set-string" "defaultBackend.image.repository=${NGINX_INGRESS_BACKEND_REPO}")
 fi
 
+# If MetalLB is installed, use a LoadBalancer for ingress, otherwise use a NodePort. Unless someone remove the dynamic flag.
+if ! helm status metallb >/dev/null 2>&1; then
+	sed -i 's/type: .*## DYNAMIC deploy_ingress.sh/    type: NodePort ## DYNAMIC deploy_ingress.sh/g' config/helm/ingress.yml
+else
+	sed -i 's/type: .*## DYNAMIC deploy_ingress.sh/    type: LoadBalancer ## DYNAMIC deploy_ingress.sh/g' config/helm/ingress.yml
+fi
```
[the remark this answers is not in the document]
Comment (supertetelman) [reply scripts/k8s/deploy_ingress.sh]: I hate to duplicate the config code, but this is a much cleaner implementation.

I  had figured that the user could remove the DYNAMIC flag if they didn't want it to be dynamic, but moving our example out of config altogether makes sense.

I introduced a new HELM_INGRESS_CONFIG global that can be used to override the workload example.

**Review by ajdecon — APPROVED** [review]
Updates LGTM!

**Pull request merged**